code/model/modelanimation.cpp

```diff
@@ -213,70 +329,219 @@ namespace animation {
+			if (!strstr(namelower, "engine") && !strstr(namelower, "radar") && strstr(namelower, "turret")) {
+				auto rotBase = std::shared_ptr<ModelAnimationSegmentSetAngle>(new ModelAnimationSegmentSetAngle(angle.h));
+				auto subsysBase = std::shared_ptr<ModelAnimationSubmodelTurret>(new ModelAnimationSubmodelTurret(sp->subobj_name, false, sip->name, std::move(rotBase)));
+				anim->addSubmodelAnimation(std::move(subsysBase));
+
+				auto rotBarrel = std::shared_ptr<ModelAnimationSegmentSetAngle>(new ModelAnimationSegmentSetAngle(angle.p));
+				auto subsysBarrel = std::shared_ptr<ModelAnimationSubmodelTurret>(new ModelAnimationSubmodelTurret(sp->subobj_name, true, sip->name, std::move(rotBarrel)));
+				anim->addSubmodelAnimation(std::move(subsysBarrel));
+			}
+			else {
+				auto rot = std::shared_ptr<ModelAnimationSegmentSetPHB>(new ModelAnimationSegmentSetPHB(angle, isRelative));
+				auto subsys = std::shared_ptr<ModelAnimationSubmodel>(new ModelAnimationSubmodel(sp->subobj_name, std::move(rot)));
+				anim->addSubmodelAnimation(std::move(subsys));
+			}
 
-			auto rotBarrel = std::shared_ptr<ModelAnimationSegmentSetAngle>(new ModelAnimationSegmentSetAngle(angle.p));
-			auto subsysBarrel = std::shared_ptr<ModelAnimationSubmodelTurret>(new ModelAnimationSubmodelTurret(sp->subobj_name, true, sip->name, std::move(rotBarrel)));
-			anim->addSubmodelAnimation(std::move(subsysBarrel));
+			sip->animations.emplace(anim, anim_name_from_subsys(sp), animation::ModelAnimationTriggerType::Initial);
 		}
 		else {
-			auto rot = std::shared_ptr<ModelAnimationSegmentSetPHB>(new ModelAnimationSegmentSetPHB(angle, isRelative));
-			auto subsys = std::shared_ptr<ModelAnimationSubmodel>(new ModelAnimationSubmodel(sp->subobj_name, std::move(rot)));
-			anim->addSubmodelAnimation(std::move(subsys));
-		}
+			std::shared_ptr<ModelAnimation> anim = ModelAnimation::createAnimation();
+
+			if (type == ModelAnimationTriggerType::TurretFired) {
+				//Turret fireds won't get reset by code, so make them auto-resetting
+				anim->m_flags += Animation_Flags::Auto_Reverse;
+			}
+			
+			auto mainSegment = std::shared_ptr<ModelAnimationSegmentSerial>(new ModelAnimationSegmentSerial());
+
+			if (optional_string("+delay:")) {
+				int delayByMs;
+				stuff_int(&delayByMs);
+				auto delay = std::shared_ptr<ModelAnimationSegmentWait>(new ModelAnimationSegmentWait(((float)delayByMs) * 0.001f));
+				mainSegment->addSegment(delay);
+			}
+
+			int delayByMsReverse = -1;
+			if (optional_string("+reverse_delay:")) {
+				stuff_int(&delayByMsReverse);
+			}
+
+			angles target{ 0,0,0 };
+			bool absolute = false;
+
+			if (optional_string("+absolute_angle:")) {
+				absolute = true;
+				vec3d angle;
+
+				stuff_vec3d(&angle);
+
+				target.p = fl_radians(angle.xyz.x);
+				target.h = fl_radians(angle.xyz.y);
+				target.b = fl_radians(angle.xyz.z);
```

Comment:
Again with stuffing a vec3d (here and elsewhere): people will expect to specify pitch/bank/heading.

It might be a good idea to create a separate `stuff_angles` function.